test/Triton/reduce-i1.mlir

```diff
@@ -0,0 +1,15 @@
+//  RUN: triton-opt %s -convert-triton-to-tritongpu -convert-triton-gpu-to-llvm | FileCheck %s
+//  CHECK-NOT: st.shared.b1
+//  CHECK-NOT: ld.shared.b1
```

Comment:
nit: might be better to check that the right st.shared/ld.shared are generated?

Reply:
Added. Did you also want to remove the CHECK-NOTs? I thought it might be valuable to make sure these illegal instructions weren't generated anywhere (even if the other ones were also generated)

Reply:
sure, it's fine to keep them